lib/hasPermissions.ts

```diff
@@ -203,16 +207,57 @@ export class HasPermissionsDirectiveVisitor<
     const hasPermissionsValidateFunction: ValidateFunction<TContext> = (
       value: unknown,
       _: unknown,
-      __: unknown,
+      container: GraphQLArgument | GraphQLInputObjectType | GraphQLObjectType,
       context: TContext,
       resolverInfo: Record<string, unknown>,
       resolverSource: unknown,
       resolverArgs: Record<string, unknown>,
+      path?: Array<string>,
     ): unknown => {
+      if (
+        (location === DirectiveLocation.INPUT_FIELD_DEFINITION ||
+          location === DirectiveLocation.INPUT_OBJECT) &&
+        container instanceof GraphQLInputObjectType &&
+        path &&
+        value instanceof Array
+      ) {
+        const verify = [];
+        for (let i = 0; i < value.length; i += 1) {
+          if (
+            value[i] !==
+            container.getFields()[path[path.length - 1]].defaultValue[i]
```

Comment:
Please, do 
```
const lastField = container.getFields()[path[path.length - 1]];
for (...)
```

No need to access it on every for iteration.

On top of that, is it correct to access the `defaultValue[i]` ?

Reply:
I think is correct to access the `defaultValue[i]`, because in some cases, the `defaultValue` is a Array, and is necessary check if all the values is equals, but thinking now, maybe is good to check if the `defaultValue` is a Array before, because if don't, no need to enter the loop.

And I changed the: 
`const lastField = container.getFields()[path[path.length - 1]];`
in advance, thanks for all the help